tinygrad/shape/view.py

```diff
@@ -33,7 +33,9 @@ def create(shape:Tuple[sint, ...], strides:Optional[Tuple[sint, ...]]=None, offs
   @functools.lru_cache(maxsize=None)  # pylint: disable=method-cache-max-size-none
   def size(self): return prod([s.max if isinstance(s, Node) else s for s,st in zip(self.shape, self.strides) if st != 0])
 
-  def vars(self) -> List[Variable]: return dedup(functools.reduce(operator.add, [x.vars() for x in self.shape+self.strides+(self.offset,) if isinstance(x, Node)], []))
+  def vars(self) -> List[Variable]:
+    flatten_mask = tuple(flatten(self.mask)) if self.mask is not None else tuple()
+    return dedup(functools.reduce(operator.add, [x.vars() for x in self.shape+self.strides+(self.offset,)+flatten_mask if isinstance(x, Node)], []))
```

Comment:
Maybe use flatten here as well instead of functools.reduce ?

Reply:
flatten does like tuple because it's not Iterator, I also revert the use of flatten in flatten_mask. There's also itertools.chain, but I think these are close enough in terms of speed and readability